model-engine/model_engine_server/domain/exceptions.py

```diff
@@ -1,4 +1,62 @@
-from model_engine_server.core.domain_exceptions import DomainException
+from dataclasses import dataclass
+
+
+class DomainException(Exception):
+    """
+    Base class for exceptions thrown for domain (business logic) errors.
+    """
+
+
+class ObjectAlreadyExistsException(DomainException):
+    """
+    Thrown when the user tries to create a model with a name that already exists.
+    """
+
+
+class ObjectNotFoundException(DomainException):
+    """
+    Thrown when a required object is not found, e.g. when creating a version for a nonexistent model
+    """
+
+
+class ObjectNotAuthorizedException(DomainException):
+    """
+    Thrown when a user tries to access an object they don't own.
+    """
+
+
+class ObjectHasInvalidValueException(DomainException, ValueError):
+    """
+    Thrown when a user tries to create an object with an invalid value.
+    """
+
+
+class ObjectNotApprovedException(DomainException):
+    """
+    Thrown when a required object is not approved, e.g. for a Bundle in review.
+    """
+
+
+@dataclass
+class DockerImageNotFoundException(DomainException):
+    """
+    Thrown when a user tries to specify a custom Docker image that cannot be found.
+    """
+
+    repository: str
```

Comment:
Just want to check that `repository` refers to a full image URL, minus the image tag. e.g. it shouldn't refer to an AWS ECR repo purely by its name. In other words, it shouldn't look like `some_service`, but `<account_id>.../some_service` or `<some dockerhub prefix>.../some_service`.

Reply:
Ok discussed offline, this is fine for now. If needed in the future, we can deprecate this field and come up with something new in its stead, like `image_url`.